pkg/agent/cniserver/pod_configuration_test.go

```diff
@@ -0,0 +1,464 @@
+				assert.Equal(t, key, podIfName)
+			} else {
+				require.Equal(t, 0, configurator.unreadyPortQueue.Len())
+			}
+		})
+	}
+}
+
+func TestProcessPortStatusMessage(t *testing.T) {
+	validOFPort := int32(1)
+	invalidOFPort := int32(0)
+	for _, tc := range []struct {
+		name            string
+		status          *openflow15.PortStatus
+		ovsPortName     string
+		ifaceInStore    bool
+		expEnqueue      bool
+		expOFportNumber *int32
+	}{
+		{
+			name: "Add OF port if port status is live",
+			status: &openflow15.PortStatus{
+				Desc: openflow15.Port{
+					PortNo: 1,
+					Length: 72,
+					Name:   []byte(podIfName),
+					State:  openflow15.PS_LIVE,
+				},
+			},
+			ovsPortName:     podIfName,
+			ifaceInStore:    true,
+			expEnqueue:      true,
+			expOFportNumber: &validOFPort,
+		}, {
+			name: "Add OF port with suffix in name",
+			status: &openflow15.PortStatus{
+				Desc: openflow15.Port{
+					PortNo: 1,
+					Length: 72,
+					Name:   []byte(fmt.Sprintf("%s\x00", podIfName)),
+					State:  openflow15.PS_LIVE,
+				},
+			},
+			ovsPortName:     podIfName,
+			ifaceInStore:    true,
+			expEnqueue:      true,
+			expOFportNumber: &validOFPort,
+		}, {
+			name: "Ignore OF port if port is not live",
+			status: &openflow15.PortStatus{
+				Desc: openflow15.Port{
+					PortNo: 1,
+					Length: 72,
+					Name:   []byte(fmt.Sprintf("%s\x00", podIfName)),
+					State:  openflow15.PS_LINK_DOWN,
+				},
+			},
+			ovsPortName:     podIfName,
+			ifaceInStore:    true,
+			expEnqueue:      false,
+			expOFportNumber: &invalidOFPort,
+		}, {
+			name: "Not enqueue OF port status message if the interface config does not exist",
+			status: &openflow15.PortStatus{
+				Desc: openflow15.Port{
+					PortNo: 1,
+					Length: 72,
+					Name:   []byte(podIfName),
+					State:  openflow15.PS_LIVE,
+				},
+			},
+			ovsPortName:     podIfName,
+			ifaceInStore:    false,
+			expEnqueue:      false,
+			expOFportNumber: nil,
+		},
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+			queue := workqueue.NewTypedDelayingQueueWithConfig[string](
+				workqueue.TypedDelayingQueueConfig[string]{
+					Name: "podMonitor",
+				})
+			podCfg := &podConfigurator{
+				ifaceStore:       interfacestore.NewInterfaceStore(),
+				statusCh:         make(chan *openflow15.PortStatus),
+				unreadyPortQueue: queue,
+				containerAccess:  newContainerAccessArbitrator(),
+			}
+			defer podCfg.unreadyPortQueue.ShutDown()
+
+			if tc.ifaceInStore {
+				podCfg.ifaceStore.AddInterface(&interfacestore.InterfaceConfig{
+					InterfaceName: podIfName,
+					ContainerInterfaceConfig: &interfacestore.ContainerInterfaceConfig{
+						PodNamespace: testPodNamespace,
+						PodName:      testPodNameA,
+						ContainerID:  podInfraContainerID,
+					},
+					OVSPortConfig: &interfacestore.OVSPortConfig{
+						PortUUID: "test-port-uuid",
+					},
+					IPs: podIPs,
+					MAC: podMac,
+				})
+			}
+
+			podCfg.processPortStatusMessage(tc.status)
+			time.Sleep(600 * time.Millisecond)
```

Comment:
ditto, this extends the execution time a lot, and why a sleep is needed if everything is synchronous.

Reply:
updated.